### DISCUSSION TITLE
How to add underdots to Chinese text?

### DISCUSSION THREAD

**Discussion by wdscxsj** [typst/typst#4682] [Q&A] [answered]
Underdots are commonly used as stress symbols in Chinese. Is it possible to get this effect? Thanks.

![Screenshot382](https://github.com/user-attachments/assets/83099727-6742-4698-967a-5bd308cc7243)

**Comment by memeplex** [comment, marked as answer]
Here are a couple of quick solutions that come to my mind, probably there are much better ones and probably there is some Chinese specific thing for this.

```typst

#let dotted1(c) = context {
  let (width: wd, height: hd) = measure(".")
  box(place(dx: .5 * wd, dy: -.7 * hd, "."))
  c
}

#let dotted2(c) = context {
  let (width: wd, height: hd) = measure(".")
  box(stack(c, v(-.7 * hd), h(.5 * wd) + ".", v(-.3 * hd)))
}

a#dotted1[b]c

a#dotted2[b]c
```

<img width="57" alt="image" src="https://github.com/user-attachments/assets/a8f7877c-4fcc-45cf-80fd-1c298e896684">

You can experiment with `math.attach` also, but I think it's problematic because of https://github.com/typst/typst/issues/366. For example:

```typst
#let dotted3(c) = $attach(limits(upright(#c)), b: inline(dot))$

a#dotted3[b]b
```

<img width="57" alt="image" src="https://github.com/user-attachments/assets/3bcf27e8-b9aa-4c3e-8d0b-5cbf27f36d6a">


Look how it's not the same "b".

Then again, try to find out whether there is something more specific for Chinese instead of these hacks, if it is so common to stress symbols like that there should be some kind of accent.
